config/configauth/configauth.go

```diff
@@ -29,13 +29,13 @@ var (
 
 // Authentication defines the auth settings for the receiver
 type Authentication struct {
-	// Authenticator specifies the name of the extension to use in order to authenticate the incoming data point.
+	// ServerAuth specifies the name of the extension to use in order to authenticate the incoming data point.
 	AuthenticatorName string `mapstructure:"authenticator"`
 }
 
 // GetAuthenticator attempts to select the appropriate from the list of extensions, based on the requested extension name.
 // If an authenticator is not found, an error is returned.
-func GetAuthenticator(extensions map[config.ComponentID]component.Extension, requested string) (Authenticator, error) {
+func GetAuthenticator(extensions map[config.ComponentID]component.Extension, requested string) (ServerAuth, error) {
```

Comment:
I would name it as `ServerAuthenticator`. Auth is typically short for `Authentication`, which is slightly different from what the type is: we are doing Auth by, among other things, calling an Authenticator.

Reply:
Agreed ^